packages/cipherstash-proxy/src/config/database.rs

```diff
@@ -0,0 +1,91 @@
+use std::{fmt::Display, time::Duration};
+
+use rustls_pki_types::ServerName;
+use serde::Deserialize;
+
+use crate::error::{ConfigError, Error};
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct DatabaseConfig {
+    #[serde(default = "DatabaseConfig::default_host")]
+    pub host: String,
+
+    #[serde(default = "DatabaseConfig::default_port")]
+    pub port: u16,
+
+    pub name: String,
+    pub username: String,
+    pub password: String,
+
+    #[serde(default = "DatabaseConfig::default_connection_timeout")]
+    pub connection_timeout: u64,
+
+    #[serde(default)]
+    pub with_tls_verification: bool,
+
+    #[serde(default = "DatabaseConfig::default_config_reload_interval")]
+    pub config_reload_interval: u64,
+
+    #[serde(default = "DatabaseConfig::default_schema_reload_interval")]
+    pub schema_reload_interval: u64,
+}
+
+impl DatabaseConfig {
+    pub fn default_host() -> String {
+        "127.0.0.1".to_string()
+    }
```

Comment:
Could this return a `&'static str`?

Reply:
Has to be turned into a String anyway to be set on the struct field.